presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/WindowFilterPushDown.java

```diff
@@ -165,6 +165,25 @@ public PlanNode visitFilter(FilterNode node, RewriteContext<Constraint> context)
         }
     }
 
+    private static boolean shouldAddFilterNode(Expression predicate)
+    {
+        if (!(predicate instanceof ComparisonExpression)) {
+            return false;
+        }
+
+        ComparisonExpression expression = (ComparisonExpression) predicate;
+        if ((expression.getLeft() instanceof QualifiedNameReference && expression.getRight() instanceof Literal)
+                || (expression.getLeft() instanceof Literal && expression.getRight() instanceof QualifiedNameReference)) {
+            Optional<Literal> literal = WindowLimitExtractor.extractLiteral(expression);
+            if (literal.isPresent() && expression.getType() == ComparisonExpression.Type.EQUAL
+                    && WindowLimitExtractor.extractValue(literal.get()) != 1) {
```

Comment:
As @electrum said, for "rn = 1"  we just need to change to using TopNRowNumberNode and no need to add a filter node on it.